File: leetcode_py/Implement strStr().py
class Solution:
    def find(self, a, b):
        for i in range(len(b)):
            if a[i] != b[i]:
                return False
        return True

    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        if len(needle) == 0:
            return 0
        for idx in range(len(haystack) - len(needle) + 1):
            if haystack[idx] == needle[0]:
                if self.find(haystack[idx:], needle):
                    return idx
        return -1

File: leetcode_js/Implement strStr().js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
	if(haystack === needle) return 0;
	if(haystack.length === 0) return -1;
	var fCheck = function(i){
		for(var j = 0 ; j < needle.length ; j++){
			if(haystack[j+i] !== needle[j]) return false;
		}
		return true;
	}
	for(var i = 0 ; i < haystack.length - needle.length + 1 ; i++){
		if(fCheck(i)) return i;
	}    
	return -1;
};